File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/ConsoleIn.cs
using System;
namespace ArqLibrarianClassic
{
    public class ConsoleIn : UserIn
    {
        public string ReadLine()
        {
            var result = Console.ReadLine().Trim();
            // Console in buffers enter, so we swallow it
            Console.ReadKey();
            return result;
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/ConsoleIn.java
package bnsit.ms.arq;

import java.util.Scanner;

public class ConsoleIn implements UserIn {
    private Scanner scanner = new Scanner(System.in);

    @Override
    public String readLine() {
        String result = scanner.nextLine().trim();
        // Console in buffers enter, so we swallow it
        // scanner.nextLine().trim()
        return result;
    }
}
